modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/jackson/databind/ObjectMapperProviderUtil.java

```diff
@@ -0,0 +1,116 @@
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleFilterProvider;
+import com.fasterxml.jackson.databind.util.ISO8601DateFormat;
+
+import com.liferay.petra.function.UnsafeSupplier;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.util.PropsValues;
+import com.liferay.portal.vulcan.jaxrs.serializer.JSONArrayStdSerializer;
+import com.liferay.portal.vulcan.jaxrs.serializer.JSONObjectStdSerializer;
+import com.liferay.portal.vulcan.jaxrs.serializer.UnsafeSupplierJsonSerializer;
+
+/**
+ * @author Alejandro Tardín
+ */
+public class ObjectMapperProviderUtil {
+
+	public static ObjectMapper getBatchEngineObjectMapper() {
+		if (PropsValues.JSON_STRING_MAX_LENGTH != _jsonStringMaxLength) {
+			_jsonStringMaxLength = PropsValues.JSON_STRING_MAX_LENGTH;
+
+			_batchEngineObjectMapper = new ObjectMapper(_getJsonFactory()) {
+				{
+					disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+					enable(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY);
+					registerModule(
+						new SimpleModule() {
+							{
+								addSerializer(
+									(Class<UnsafeSupplier<Object, Exception>>)
+										(Class<?>)UnsafeSupplier.class,
+									new UnsafeSupplierJsonSerializer());
+							}
+						});
+					setDateFormat(new ISO8601DateFormat());
+					setSerializationInclusion(JsonInclude.Include.NON_NULL);
+				}
+			};
+		}
+
+		return _batchEngineObjectMapper;
+	}
+
+	public static ObjectMapper getObjectMapper() {
+		if (PropsValues.JSON_STRING_MAX_LENGTH != _jsonStringMaxLength) {
+			_jsonStringMaxLength = PropsValues.JSON_STRING_MAX_LENGTH;
+
+			_objectMapper = new ObjectMapper(_getJsonFactory()) {
+				{
+					configure(
+						MapperFeature.SORT_PROPERTIES_ALPHABETICALLY, true);
+					enable(SerializationFeature.INDENT_OUTPUT);
+					registerModule(
+						new SimpleModule() {
+							{
+								addSerializer(
+									JSONArray.class,
+									new JSONArrayStdSerializer(
+										JSONArray.class));
+								addSerializer(
+									JSONObject.class,
+									new JSONObjectStdSerializer(
+										JSONObject.class));
+								addSerializer(
+									(Class<UnsafeSupplier<Object, Exception>>)
+										(Class<?>)UnsafeSupplier.class,
+									new UnsafeSupplierJsonSerializer());
+							}
+						});
+					setDateFormat(new ISO8601DateFormat());
+					setFilterProvider(
+						new SimpleFilterProvider() {
+							{
+								addFilter(
+									"Liferay.Vulcan",
+									SimpleBeanPropertyFilter.serializeAll());
+							}
+						});
+					setSerializationInclusion(JsonInclude.Include.NON_ABSENT);
+				}
+			};
+		}
+
+		return _objectMapper;
+	}
+
+	private static JsonFactory _getJsonFactory() {
+		return new JsonFactoryBuilder(
+		).streamReadConstraints(
+			StreamReadConstraints.builder(
+			).maxStringLength(
+				PropsValues.JSON_STRING_MAX_LENGTH
```

Comment:
Maybe better passing the value as an argument to avoid reading again the value just in case it takes less time (as we already have read the value)

Reply:
I saw you did it your self. :+1: